metadata-io/src/main/java/com/linkedin/metadata/timeline/differ/SchemaDiffer.java

```diff
@@ -173,109 +178,60 @@ private static ChangeEvent getDescriptionChange(SchemaField baseField, SchemaFie
+        // Assumes that two different fields on the same schema would not have the same description, terms,
+        // or tags and share the same type
+        SchemaField renamedField = findRenamedField(curBaseField, targetFields, renamedFields);
+        if (renamedField == null) {
+          processRemoval(changeCategory, changeEvents, datasetUrn, curBaseField);
           ++baseFieldIdx;
-          ++targetFieldIdx;
-        } else if (comparison < 0) {
-          // BaseFiled got removed. Non-backward compatible change + Major version bump
-          if (ChangeCategory.TECHNICAL_SCHEMA.equals(changeCategory)) {
-            changeEvents.add(ChangeEvent.builder()
-                .category(ChangeCategory.TECHNICAL_SCHEMA)
-                .elementId(getSchemaFieldUrn(datasetUrn, curBaseField))
-                .target(datasetUrn.toString())
-                .changeType(ChangeOperation.REMOVE)
-                .semVerChange(SemanticChangeType.MAJOR)
-                .description(BACKWARDS_INCOMPATIBLE_DESC + "removal of the field'" + getFieldPathV1(curBaseField) + "'.")
-                .build());
-          }
+        } else {
+          changeEvents.add(generateRenameEvent(datasetUrn, curBaseField, renamedField));
           List<ChangeEvent> propChangeEvents = getFieldPropertyChangeEvents(curBaseField, curTargetField, datasetUrn,
               changeCategory);
           changeEvents.addAll(propChangeEvents);
           ++baseFieldIdx;
+          renamedFields.add(renamedField);
+        }
+      } else {
+        // The targetField got added or a rename occurred. Forward & backwards compatible change + minor version bump.
+        SchemaField renamedField = findRenamedField(curTargetField, baseFields, renamedFields);
+        if (renamedField == null) {
+          processAdd(changeCategory, changeEvents, datasetUrn, curTargetField);
+          ++targetFieldIdx;
         } else {
-          // The targetField got added. Forward & backwards compatible change + minor version bump.
-          if (ChangeCategory.TECHNICAL_SCHEMA.equals(changeCategory)) {
-            changeEvents.add(ChangeEvent.builder()
-                .category(ChangeCategory.TECHNICAL_SCHEMA)
-                .elementId(getSchemaFieldUrn(datasetUrn, curTargetField))
-                .target(datasetUrn.toString())
-                .changeType(ChangeOperation.ADD)
-                .semVerChange(SemanticChangeType.MINOR)
-                .description(
-                    BACK_AND_FORWARD_COMPATIBLE_DESC + "the newly added field '" + getFieldPathV1(curTargetField) + "'.")
-                .build());
-          }
+          changeEvents.add(generateRenameEvent(datasetUrn, renamedField, curTargetField));
           List<ChangeEvent> propChangeEvents = getFieldPropertyChangeEvents(curBaseField, curTargetField, datasetUrn,
               changeCategory);
           changeEvents.addAll(propChangeEvents);
           ++targetFieldIdx;
+          renamedFields.add(renamedField);
```

Comment:
Just an observation - this method is a bit long. Pretty hard for an unfamiliar reader to follow it